#! /usr/bin/env python

PACKAGE='ece6460_pointcloud_example'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                        Type       Lvl    Description                                          Default     Min      Max
gen.add("x_min",                    double_t,  0,     "Region of interest",                                -100.0,     -1000.0, 0.0)
gen.add("x_max",                    double_t,  0,     "Region of interest",                                100.0,      0.0,     1000.0)
gen.add("y_min",                    double_t,  0,     "Region of interest",                                -100.0,     -1000.0, 0.0)
gen.add("y_max",                    double_t,  0,     "Region of interest",                                100.0,      0.0,     1000.0)
gen.add("z_min",                    double_t,  0,     "Region of interest",                                -100.0,     -1000.0, 0.0)
gen.add("z_max",                    double_t,  0,     "Region of interest",                                3.0,        0.0,     1000.0)
gen.add("voxel_size",               double_t,  0,     "Downsampling voxel size",                           0.3,        0.01,    1.0)
gen.add("num_normal_neighbors",     int_t,     0,     "Number of neighbors to search to estimate normals", 50,         5,       100)
gen.add("cluster_tol",              double_t,  0,     "Distance to include point in cluster",              0.5,        0.001,   10.0)
gen.add("min_cluster_size",         int_t,     0,     "Minimum allowed number of points in a cluster",     5,          1,       1000)
gen.add("max_cluster_size",         int_t,     0,     "Maximum allowed number of points in a cluster",     500,        100,     10000)

exit(gen.generate(PACKAGE, PACKAGE, "PointCloudExample"))
